utils/validate_images.js

```diff
@@ -18,19 +20,37 @@ const ALLOWED_IMG_EXT = ['.png', '.jpeg', '.jpg', '.svg'];
  * Validate all folders contain no more than MAX_NUM_IMG images
  * @param {Array} files - The array of globbed file names
  */
-const validateImageCounts = (files) => {
-  const directories = files
-    .filter((file) => isDirectory(file))
-    .filter((folder) => {
-      return getImageCount(folder) > MAX_NUM_IMG;
+
+const validateImageCounts = (packDirs) => {
+  const directories = packDirs
+    .map((pack) => {
+      const packDirName = path.dirname(pack);
+      // get all images for pack
+      const imagePaths = glob.sync(
+        path.resolve(path.dirname(pack), '**/*.+(png|jpeg|jpg|svg)')
+      );
+      const packConfig = readPackFile(pack).contents[0];
+      const iconPath = packConfig.icon
+        ? path.resolve(path.dirname(pack), packConfig.icon)
+        : null;
+      const logoPath = packConfig.logo
+        ? path.resolve(path.dirname(pack), packConfig.logo)
+        : null;
+
+      const screenshotPaths = imagePaths.filter(
+        (p) => p !== iconPath && p !== logoPath
+      );
+
+      if (screenshotPaths.length > MAX_NUM_IMG) {
+        return {
+          folder: packDirName,
+          imageCount: screenshotPaths.length,
+        };
+      }
     })
-    .map((folder) => {
-      return {
-        folder,
-        imageCount: getImageCount(folder),
-      };
-    });
-  if (directories.length > 0) {
+    .filter(Boolean);
```

Comment:
what does this filter do?

Reply:
the map will loop over the packs and make an entry for each one even if that entry is undefined (in the case that the image count is not exceeding 6) so this filter quickly drops any item in the array that's falsy, all the undefineds